### DISCUSSION TITLE
armv6/7 packaged version

### DISCUSSION THREAD

**Discussion by sl8vz** [zwave-js/zwave-js-ui#2873] [Q&A] [answered]
Hi,

Would it be possible to include a pre-built version for arm 32 bits platform ?
This would be very helpful since in general this concerns lower-end systems on which running containers can be difficult and which don't necessarily ship yarn.

I'd be happy to help/test if needed. I guess you use qemu dockers for building ?

Thanks,
Sebastien

**Comment by robertsLando** [comment]
Actually I'm not using quemu but custom runners, I don't have a custom runner setup for armv6/7 yet

**Reply by sl8vz** [reply]
Hi,
FYI I managed to package an armv7 build using the same steps as you do in `package_arm64.yml` and it works like a charm.
I've used qemu with `node:16.19-bullseye-slim` with platform `linux/arm/v7` on my machine.
`yarn install` complains with some oom issues but after few tries it successfully ends.
Not sure how to properly do that with github actions, [run on architecture](https://github.com/marketplace/actions/run-on-architecture) might be useful.

**Reply by robertsLando** [reply]
It should work by using the gh action you suggested and mapping the project directory into the docker conatiner, I need to test it out BTW

**Reply by sl8vz** [reply]
The only issues is the node version of the supported images.  Do you have a hard req on node 16 ?

**Reply by robertsLando** [reply]
Nope it's just more tested, why is that an issue?

**Reply by robertsLando** [reply]
@sl8vz Did you tried to run the package in armv7? Does it works?

**Reply by sl8vz** [reply]
> Nope it's just more tested, why is that an issue?`

containers supported by `run on architecture`  have lesser version in general.

> Did you tried to run the package in armv7? Does it works?

Yes it runs fine.

**Reply by robertsLando** [reply]
Did you have to build nodejs or did pkg fetched the binaries?

**Comment by sl8vz** [comment]
It had to build node. Took a long time.

Le ven. 20 janv. 2023, 17:46, Daniel Lando ***@***.***> a
écrit :

**Reply by robertsLando** [reply]
> It had to build node. Took a long time.

Ok that's what I thought... In this case I cannot use GH actions as them will fail after 6h. The solution would be to get the builded node and inject it so it will skip the build...

**Comment by robertsLando** [comment]
@sl8vz Could you give a try to it? Check armv7 asset https://github.com/zwave-js/zwave-js-ui/releases/tag/v8.8.0

**Reply by sl8vz** [reply]
Hi @robertsLando ,
I had a quick try on docker/qemu but unfortunately it doesn't properly starts.
```
-rw-r--r--.   1 root root  57182667 Jan 23 16:51 zwave-js-ui-v8.8.0-linux-armv7.zip
# chmod +x ./zwave-js-ui
# ./zwave-js-ui
Error: Dynamic loading not supported
    at process.dlopen (pkg/prelude/bootstrap.js:2255:28)
    at Object.Module._extensions..node (node:internal/modules/cjs/loader:1189:18)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Module.require (node:internal/modules/cjs/loader:1005:19)
    at Module.require (pkg/prelude/bootstrap.js:1855:31)
    at require (node:internal/modules/cjs/helpers:102:18)
    at load (/snapshot/zwave-js-ui/node_modules/node-gyp-build/index.js:22:10)
    at Object.<anonymous> (/snapshot/zwave-js-ui/node_modules/@serialport/bindings-cpp/dist/load-bindings.js:10:46)
    at Module._compile (pkg/prelude/bootstrap.js:1930:22)
```

**Reply by robertsLando** [reply]
Ok so my hacky hack didn't worked :( seems linuxstatic doesn't support node native addons (needed for serialport node)

Will try to see if I find another way to do this

**Comment by robertsLando** [comment, marked as answer]
Finally. DONE: https://github.com/zwave-js/zwave-js-ui/releases/tag/v8.8.5 🚀

**Reply by sl8vz** [reply]
Wow that's a very cool news !
Many thanks @robertsLando
